proposals/1/patch-steps.md

```diff
@@ -211,8 +228,64 @@ declare type PatchStepInclude = {
 	// File Path, default protocol "mod:"
 	"src": string;
 };
+
+
+/*
+ * The FOR_IN patch step takes a value entry from the values property,
+ * goes through the body statements,
+ * clones and replaces keyword match with the current value entry inside the statement object,
+ * and then executes the current statement.
+ * Note: If the keyword is a string, then the values must be an array of strings.
+ *       If the keyword is a PatchStepObjectMatch, then the values must be an array of PatchStepObjects.
+ */
+declare type PatchStepForIn = {
+	"type": "FOR_IN";
+	"values": string[] | PatchStepObject[];
+	// interpreted as a Regular Expression
+	"keyword": string | PatchStepObjectMatch;
```

Comment:
These two values seem to have their types bound together... Probably not a problem since the notes point this out, IDK.

Reply:
It's intentional done.